source-code/core/src/lint/rules/additionalKey.ts

```diff
@@ -0,0 +1,24 @@
+import type { Context } from '../context.js'
+import { createRule } from '../rule.js'
+
+export const additionalKeyRule = createRule('inlang.additionalKey', () => {
+	let context: Context
+	let referenceLanguage: string
+
+	return {
+		initialize: (config) => {
+			context = config.context
+			referenceLanguage = config.referenceLanguage
+		},
+		visitors: {
+			Resource: ({ target }) => {
+				if (target && target.languageTag.name === referenceLanguage) return 'skip'
+			},
+			Message: ({ target, reference }) => {
+				if (!reference && target) {
+					context.report(target, `Message with id '${target.id.name}' is specified, but missing in the reference`)
+				}
+			},
```

Comment:
This seems like a great API. 

Super simple if reference does not contain target report something on `target`. I would have expected to report the level here too though: `context.report("warn", "Message with ...")`. The user could "overwrite" the level in the lint config.

Reply:
> I would have expected to report the level here too though: `context.report("warn", "Message with ...")`. The user could "overwrite" the level in the lint config.

This already happens. If the `rule` author would specify the `level` on each message, then how would we be able to tell which one the user wants to override? A `rule` can only report a single `level`. The `rule` author can set a default level, but a user can override it.